datafusion/src/physical_plan/sorts/sort_preserving_merge.rs

```diff
@@ -280,7 +287,11 @@ pub(crate) struct SortPreservingMergeStream {
     ///
     /// Exhausted cursors will be popped off the front once all
     /// their rows have been yielded to the output
-    cursors: Vec<VecDeque<Arc<SortKeyCursor>>>,
+    batches: Vec<VecDeque<Arc<RecordBatch>>>,
```

Comment:
Mechanically the changes in this PR make sense to me, but I don't quite follow how the elements in `batches` correspond to the cursors in `min_heap` (especially the code that drains all by the last `RecordBatch` [below](https://github.com/apache/arrow-datafusion/pull/1624/files#diff-0cd58001ad86fb06e8667fdac3d89409b4624fb2a21c882484beeaf00c9f5df7R528-R531).

Reply:
I wonder if the `Arc<RecordBatch>` could be removed from `SortKeyCursor` and therefore this just be `Vec<VecDequeue<RecordBatch>>`. `SortKeyCursor::batch` only appears to now be used for debug formatting